hw_1/Task1.py

```diff
@@ -1,30 +1,27 @@
 import random
 
 # Create a list of 100 random numbers from 0 to 1000
-my_list = [random.randint(0, 1000) for _ in range(100)]
+lst = [random.randint(0, 1000) for _ in range(100)]
 
 # Sort the list from min to max (without using sort())
-my_sorted_list = []
-for x in range(len(my_list)):
-    min_value = my_list[0]
-    for y in range(len(my_list) - 1):
-        if my_list[y + 1] < min_value:
-            min_value = my_list[y + 1]
-    my_sorted_list.append(min_value)
-    my_list.remove(min_value)
+n = len(lst)
+for i in range(n - 1):
+    for j in range(n - i - 1):
+        if lst[j] > lst[j + 1]:
+            lst[j], lst[j + 1] = lst[j + 1], lst[j]
 
 # Calculate average for even and odd numbers
 average_even = 0
 average_odd = 0
 even_counter = 0
 odd_counter = 0
 
-for x in range(len(my_sorted_list)):
-    if x % 2 == 0:
-        average_odd += my_sorted_list[x]
+for i, item in enumerate(lst):
```

Comment:
Enumerate allows iterate without indexing, so there will be better

```
for _, item in enumerate(lst):
if item % 2 == 0:
        average_odd += item
```

Reply:
Or you can get odd/even list with list comprehensive and count average with `sum (), len()` functions

`odd_list = [i for i in generated_list if i % 2]`